volatility3/framework/plugins/windows/etwpatch.py

```diff
@@ -0,0 +1,128 @@
+# This file is Copyright 2019 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+import logging
+
+from volatility3.framework import exceptions, interfaces, renderers
+from volatility3.framework.configuration import requirements
+from volatility3.framework.renderers import format_hints
+from volatility3.plugins.windows import pslist, pe_symbols 
+
+vollog = logging.getLogger(__name__)
+
+class EtwPatch(interfaces.plugins.PluginInterface):
+    """Detects ETW patching by examining the first opcode of EtwEventWrite in ntdll.dll."""
+
+    _version = (1, 0, 0)
+    _required_framework_version = (2, 26, 0)
+
+    @classmethod
+    def get_requirements(cls):
+        return [
+            requirements.ModuleRequirement(
+                name='kernel',
+                description='Windows kernel',
+                architectures=["Intel32", "Intel64"]
+            ),
+            requirements.VersionRequirement(
+                name="pslist", component=pslist.PsList, version=(3, 0, 0)
+            ),
+            requirements.VersionRequirement(
+                name="pe_symbols", component=pslist.PsList, version=(3, 0, 0)
+            ),
+            requirements.ListRequirement(
+                name='pid',
+                description='Filter on specific process IDs',
+                element_type=int,
+                optional=True
+            )
+        ]
+
+    def _get_dll_vads(self, proc, dll_name):
+        """Retrieve VADs for a specific DLL in the process."""
+        collected_modules = pe_symbols.PESymbols.get_proc_vads_with_file_paths(proc)
+        return [
+            (vad_start, vad_size, proc.add_process_layer())
+            for vad_start, vad_size, filepath in collected_modules
+            if pe_symbols.PESymbols.filename_for_path(filepath) == dll_name
+        ]
+
+    def _find_symbols(self, dll_name, symbols, proc_layer_name, dll_vads):
+        """Find symbols for a specific DLL."""
+        filter_module = {dll_name: {"names": symbols}}
```

Comment:
Instead of the raw "names" string use the identifier constant like unhooked system calls does here:

https://github.com/volatilityfoundation/volatility3/blob/develop/volatility3/framework/plugins/windows/unhooked_system_calls.py#L29